stl/inc/xutility

```diff
@@ -4345,6 +4359,38 @@ _BidIt2 move_backward(_ExPo&&, _BidIt1 _First, _BidIt1 _Last, _BidIt2 _Dest) noe
 #endif // _HAS_CXX17
 
 // FUNCTION TEMPLATE fill
+
+// _Iterator_is_contiguous<_Iter> reports whether an iterator is known to be contiguous.
+// (Without concepts, this detection is limited, which will limit when we can activate optimizations.)
+
+#ifdef __cpp_lib_concepts
+// When concepts are available, we can detect arbitrary contiguous iterators.
+template <class _Iter>
+inline constexpr bool _Iterator_is_contiguous = contiguous_iterator<_Iter>;

```

Comment:
Note that this is still affected by the volatile pointer Bug, which leads to `volatile int*` not being contiguous

Reply:
I don't thinks it's a problem since volatile types are excluded from any optimizations

Reply:
Agreed, shouldn't be a problem here. Thanks for mentioning it, I had forgotten how the concept behaved for volatile.